src/components/scenes/SendScene.tsx

```diff
@@ -151,14 +153,23 @@ class SendComponent extends React.PureComponent<Props, State> {
     }
   }
 
-  updateSendConfirmationTx = (guiMakeSpendInfo: GuiMakeSpendInfo) => {
+  updateSendConfirmationTx = (guiMakeSpendInfo: GuiMakeSpendInfo, isCheckScamWarning?: boolean) => {
```

Comment:
Nitpick: Since you're calling dismissScamWarning on component mount then it should be in componentDidMount. updateSendConfirmationTx won't ever be called with this additional isCheckScamWarning argument anywhere else and it seems the only reason to add to updateSendConfirmationTx was for the convenience of using the existing ugly block of recipientAddress identifying code. 

I would hate to see that awful ternary repeated again so a compromise would be to rename isCheckScamWarning to reflect its actual use as a one time on-mount flag.

Reply:
I was trying to avoid re-checking the recipientAddress but I can move things around

Reply:
Yeah initially I tried to factor out that ternary that was parsing the address so I could do as you described, and found that it was similar to handleAddressFromScan but not quite (missing the check for the GuiMakeSpendInfo publicAddress). Ugly, so I went with what I thought was the lesser of the evils.

I'll just rename